homeassistant/components/flux/switch.py

```diff
@@ -193,12 +226,49 @@ def __init__(
         mode,
         interval,
         transition,
-    ):
+    )
+
+    async_add_entities([flux])
+
+    platform = entity_platform.async_get_current_platform()
+
+    async def async_update() -> None:
+        """Update lights."""
+        await flux.async_flux_update()
+
+    platform.async_register_entity_service(
+        "flux_update",
+        {},
+        async_update,
+    )
+
+
+class FluxSwitch(SwitchEntity, RestoreEntity):
+    """Representation of a Flux switch."""
+
+    def __init__(
+        self,
+        name: str,
+        hass: HomeAssistant,
+        lights,
+        start_time: time | None,
+        sunset_time: time | None,
+        stop_time: time | None,
+        start_colortemp,
+        sunset_colortemp,
+        stop_colortemp,
+        brightness,
+        disable_brightness_adjust,
+        mode,
+        interval: timedelta,
+        transition: timedelta,
+    ) -> None:
```

Comment:
Maybe a dataclass would make sense?

Reply:
The use of a dataclass does seem logical. However, I question if this merge request is the right moment to introduce it. The second tip on [this page](https://developers.home-assistant.io/docs/review-process/#creating-the-perfect-pr) advises against making small unrelated improvements.

Reply:
A dataclass makes sense, but it should indeed happen in a follow-up PR, not in this PR 👍